ARCHITECTURE.MD

```diff
@@ -54,61 +54,212 @@ TOPP consists of comprehensive suite of command-line tools built on top of the O
   - Identification result visualization
   - Interactive data analysis
 
-### 4. Scripting  
-- **Scripting & API: pyOpenMS** 
+### 4. Scripting
+- **Scripting & API: pyOpenMS**
   - Python bindings that expose core functionalities
-  - Rapid prototyping of algorithms in Python.
-  -Integration with other scientific Python libraries (e.g., NumPy, SciPy).
-  - Development of custom data processing workflows.
-  - Interactive data analysis and visualization in Jupyter notebooks.
+  - Rapid prototyping of algorithms in Python
+  - Integration with other scientific Python libraries (e.g., NumPy, SciPy)
+  - Development of custom data processing workflows
+  - Interactive data analysis and visualization in Jupyter notebooks
 
-## Data flowchart
-
-```mermaid	
+#### pyOpenMS Generation with Autowrap
+  
+The Python bindings for OpenMS are generated through a specialized process:
+
+- **Autowrap Tool**: A custom Python tool developed specifically for generating OpenMS bindings
+- **Declaration Process**:
+  - Developers declare classes and functions to be exposed to Python in `.pxd` files located in the `pxds/` directory
+  - These declarations specify the function signatures and class hierarchies
+- **Code Generation Workflow**:
+  1. Autowrap reads the declarations and automatically generates wrapping code
+  2. Manual wrapper code can be added in the `addons/` folder for functionality that can't be wrapped automatically
+  3. The output is a `pyopenms.pyx` file in the `pyopenms/` directory
+  4. Cython translates this `.pyx` file to C++ code (`pyopenms.cpp`)
+  5. A C++ compiler compiles this code into a Python module importable via `import pyopenms`
+- **Customization Points**:
+  - Special annotations like `wrap-ignore`, `wrap-as`, `wrap-iter-begin`, etc. can be used to customize the wrapping process
+  - Allows for fine-grained control over how C++ classes and methods are exposed to Python
+
+#### Manual Extensions with PYX Files
+
+While autowrap handles most standard C++ to Python wrapping automatically, manual PYX files (in the `addons/` directory) enable advanced functionality that goes beyond simple wrapping:
+
+- **Complex C++ Features**: Handling C++ features that autowrap cannot automatically process, such as:
+  - Operator overloading (e.g., implementing Python's `__add__`, `__getitem__`)
+  - Complex template instantiations
+  - Multiple inheritance scenarios
+  - Static methods and variables
```

Comment:
Actually autowrap can do all of this. Maybe some very specific scenarios where the default wrappers are not what you want.

Reply:
true. will update.